### PR TITLE
Reversal algorithm for array rotation

### PR DESCRIPTION
**Pull request by Mohitmadhav**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added reversal algorithm for Array Rotation

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#5756

### CODE REVIEW

File: .vscode/c_cpp_properties.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (gat786) [inline .vscode/c_cpp_properties.json:1]: don't include this file in pr's its vs-code specific and doesn't require to be in the codebase.

File: C-Plus-Plus/ds/Reversal algorithm for array rotation (GSSoC'21).cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
+#include <bits/stdc++.h>
```
Comment (gat786) [inline C-Plus-Plus/ds/Reversal algorithm for array rotation (GSSoC'21).cpp:1]: rename the filename don't add gssoc or anything. Keep your file name as just the name of thing you have implemented and that too without spaces in the name.

File: C-Plus-Plus/ds/Reversal algorithm for array rotation (GSSoC'21).cpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+int main()
+{
+    int n,d;
+    cout<<"Enter array length:\n";
+    cin>>n;
+    int arr[n];
+
+    cout<<"Enter elements of the array:\n";
+    for(int i=0;i<n;i++)
+    cin>>arr[i];
```
Comment (gat786) [inline C-Plus-Plus/ds/Reversal algorithm for array rotation (GSSoC'21).cpp:53]: please use `{` for opening code blocks and `}` for closing them. While this is a working code. I personally don't encourage codes without braces.

File: C-Plus-Plus/ds/Reversal algorithm for array rotation (GSSoC'21).cpp [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+    ArrayReverseAlgo r;
+    r.leftRotate(arr,n,d);
+
+    cout<<"The rotated array is:\n";
+    r.printArray(arr,n);
+
+
+    return 0;
+}
```
Comment (gat786) [inline C-Plus-Plus/ds/Reversal algorithm for array rotation (GSSoC'21).cpp:66]: There is no explanation of the question that you are trying to solve. There's no space or time complexity added. There is also no example of running this code which is added here.

Comment (gat786) [reply C-Plus-Plus/ds/Reversal algorithm for array rotation (GSSoC'21).cpp:66]: You haven't added any comments in the code either which makes this code highly undocumented.

**Review by gat786 — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — COMMENTED** [review]
Code plagiarized from [here](https://www.geeksforgeeks.org/program-for-array-rotation-continued-reversal-algorithm/)

**Pull request closed without merging**